lightning/src/ln/msgs.rs

```diff
@@ -1161,6 +1162,9 @@ impl Writeable for Init {
 	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
 		// global_features gets the bottom 13 bits of our features, and local_features gets all of
 		// our relevant feature bits. This keeps us compatible with old nodes.
+		encode_tlv_stream!(w, {
+			(2, self.net_address, option)
+		});
 		self.features.write_up_to_13(w)?;
```

Comment:
I am not sure if this is the right way. Also what value should be put inside `encode_tlv_stream!`? Here I am using `2`
@TheBlueMatt

Reply:
Spec says the type is 3, not 2. Also it needs to be after all the other stuff. You should probably add a test for a specific encoding in the tests at the bottom of the file in `encoding_init`. Specifically, I believe if you take an existing test and append `0307017f000103e8` you should have a network address of `127.0.0.1:1000`. (type 3, length 7, network address type 1, followed by the ipv4 encoding).

Reply:
```
		assert_eq!(msgs::Init { features: InitFeatures::empty(),
			net_address: Some(msgs::NetAddress::IPv4 {
				addr: [127, 0, 0, 1],
				port: 1000,
			}),
		}.encode(), hex::decode("0307017f000103e8").unwrap());
```
Is this okay?